src/annotator/anchoring/test/types-test.js

```diff
@@ -118,16 +118,26 @@ describe('annotator/anchoring/types', () => {
   });
 
   describe('TextPositionAnchor', () => {
-    let fakePosFromRange;
-    let fakePosToRange;
+    let FakeTextRange;
 
     beforeEach(() => {
-      fakePosFromRange = sinon.stub();
-      fakePosToRange = sinon.stub();
+      const textRange = {
+        start: { element: container, offset: 0 },
+        end: { element: container, offset: 1 },
+
+        relativeTo: sinon.stub(),
+        toRange: sinon.stub(),
+      };
+
+      FakeTextRange = {
```

Comment:
I've noticed a fair amount of faking in the tests from this recent thread of PRs. It makes me wonder if we might pause and reflect about our philosophy here, such that we don't risk regressions if the faking doesn't match reality (i.e. if `TextRange`'s implementation changes). Nothing new here...I wonder if a pattern in which we only just-in-time stub the methods relevant to a given test in the test itself could be plausible. Or some sort of test tooling or linting that could assure that we only stub things we use in the test and that they're stubbed correctly (ahem, `magicMock`) Just musing...

Reply:
I'm inclined to agree in the case of tests for `range.js`. The amount of mocking does feel overkill here. I avoided changing the testing strategy in these PRs though to avoid mixing functional changes with orthogonal refactoring. We do have some integration tests for anchoring as well as type checking to help ensure that everything is in sync though.